res/values/strings.xml

```diff
@@ -1439,68 +1426,68 @@
+    <string name="preferences__if_read_receipts_are_disabled_you_wont_be_able_to_see_read_receipts">If read receipts are disabled, others won\'t be able to see if you read their messages, but you also won\'t be able to see if others have read your messages.</string>
     <string name="preferences__typing_indicators">Typing indicators</string>
-    <string name="preferences__if_typing_indicators_are_disabled_you_wont_be_able_to_see_typing_indicators">If typing indicators are disabled, you won\'t be able to see typing indicators from others.</string>
+    <string name="preferences__if_typing_indicators_are_disabled_you_wont_be_able_to_see_typing_indicators">If typing indicators are disabled, others won\'t be able to see when you are typing, but you also won\'t be able to see when others are typing.</string>
     <string name="preferences__request_keyboard_to_disable_personalized_learning">Request keyboard to disable personalized learning</string>
     <string name="preferences_app_protection__blocked_contacts">Blocked contacts</string>
     <string name="preferences_chats__when_using_mobile_data">When using mobile data</string>
     <string name="preferences_chats__when_using_wifi">When using Wi-Fi</string>
     <string name="preferences_chats__when_roaming">When roaming</string>
     <string name="preferences_chats__media_auto_download">Media auto-download</string>
-    <string name="preferences_chats__message_trimming">Message trimming</string>
+    <string name="preferences_chats__message_trimming">Message storage limits</string>
     <string name="preferences_storage__storage_usage">Storage usage</string>
     <string name="preferences_storage__photos">Photos</string>
     <string name="preferences_storage__videos">Videos</string>
     <string name="preferences_storage__files">Files</string>
     <string name="preferences_storage__audio">Audio</string>
-    <string name="preferences_storage__review_storage">Review storage</string>
-    <string name="preferences_advanced__use_system_emoji">Use system emoji</string>
+    <string name="preferences_storage__review_storage">View all stored files</string>
+    <string name="preferences_advanced__use_system_emoji">Use Android emoji</string>
     <string name="preferences_advanced__disable_signal_built_in_emoji_support">Disable Signal\'s built-in emoji support</string>
-    <string name="preferences_advanced__relay_all_calls_through_the_signal_server_to_avoid_revealing_your_ip_address">Relay all calls through the Signal server to avoid revealing your IP address to your contact. Enabling will reduce call quality.</string>
+    <string name="preferences_advanced__relay_all_calls_through_the_signal_server_to_avoid_revealing_your_ip_address">For unlisted contacts calls are always routed through the Signal server to avoid revealing your IP address to your contact. Enabling this will also relay calls with listed contacts, but it will reduce call quality.</string>
```

Comment:
I think the string I proposed is more clear because of the signal words "will ALSO relay" ", BUT it will reduce"  which shows this does something in addition to default rather than overwriting default. Also the order of the sentence as you phrased it might allow users to draw a conclusion on only the first sentence without reading further, while we want promote the user to take a decision on the full description.
Also, your text is longer than the one I proposed and consists of three sentences rather than two, both of which make your texts slightly less pleasant to read.

Hope you agree to keep this one.

Reply:
This is about "any contacts that you have saved in your phone" where "save on your phone"does not mean "saved within the Signal app" but does mean "saved within the Android contacts app".

This issue where a proper terminology is lacking stems form the annoying oversimplification Android makes when describing what a contact is: If you saved someones number on your phone, he is a contact. If you did not save his number, he is not a contact.
Obviously bullshit. I may have had contact with that person for years prior to  storing his number in my phones address book, so he was already a contact.

The same applies to using Signal: I may chat with someone on Signal for a year, but if I haven't saved his number to my android 'contacts' app I am not supposed to call him a contact.

The challenge at hand is to inform  users that it makes a difference whether or not a contact is saved to your Android contacts app.

This is what the text looks like if you do not follow the oversimplified Android terminology:
"
For _contacts who are saved to your Android contacts app_ calls are always routed through the Signal server to avoid revealing your IP address to your _conversation partner_. Enabling this will also relay calls with _contacts who are not saved to your Android contacts app_, but it will reduce call quality.
"
Now how can be simplify that without losing any information and without risking ambiguity?

Reply:
I agree with you, and I understand that is frustrating to deal with the bad design decisions on Android. But I'm not going to change that in Molly. If, in Android, a contact is an entry in the contact app, it's ok. Regarding this string, I propose to you the following: describe how it works with "contacts" vs how it works for everyone else, without actual defining what is a contact.